js/src/edit-block/components/fields-grid.js

```diff
@@ -7,196 +7,210 @@ import className from 'classnames';
+						 */
+						const selectField = ( event ) => {
+							event.stopPropagation();
+							const newSelectedField = { name: field.name };
+							if ( field.hasOwnProperty( 'parent' ) ) {
+								newSelectedField.parent = field.parent;
+							}
+
+							setSelectedField( newSelectedField );
+							setPanelDisplaying( FIELD_PANEL );
+						};
+						const isSelected = getIsSelected( field );
+						const shouldDisplayMoveButtons = fields.length > 1;
+						const isUpButtonDisabled = 0 === index;
+						const isDownButtonDisabled = index >= ( fields.length - 1 );
+						const FieldIcon = getFieldIcon( field.control );
 
-							return (
+						return (
+							<div
+								role="row"
+								className={ className(
+									{ 'is-selected': isSelected },
+									'field w-full',
+									getWidthClass( field.width )
+								) }
+								key={ `field-item-${ index }` }
+								tabIndex={ 0 }
+								aria-label={ sprintf(
+								/* translators: %1$s: the label of the field */
+									__( 'Field: %1$s', 'genesis-custom-blocks' ),
+									field.label
+								) }
+								onClick={ selectField }
+								onKeyPress={ selectField }
+							>
 								<div
-									className={ className(
-										{ 'is-selected': isSelected },
-										'field w-full',
-										getWidthClass( field.width )
-									) }
-									key={ `field-item-${ index }` }
 									role="gridcell"
-									tabIndex={ 0 }
-									aria-label={ sprintf(
-										// translators: %s: the label of the field
-										__( 'Field: %s', 'genesis-custom-blocks' ),
-										field.label
-									) }
-									onClick={ selectField }
-									onKeyPress={ selectField }
+									className="relative flex flex-col items-center w-full p-4 bg-white border border-gray-400 rounded-sm hover:border-black"
+									id={ `field-item-${ index }` }
 								>
-									<div
-										className="relative flex items-center w-full p-4 bg-white border border-gray-400 rounded-sm hover:border-black"
-										id={ `field-item-${ index }` }
-									>
-										<div>
-											<svg className="fill-current h-6 w-6" viewBox="0 0 24 24">
-												<path d="M0 0h24v24H0z" fill="none" />
-												<path d="M5 17v2h14v-2H5zm4.5-4.2h5l.9 2.2h2.1L12.75 4h-1.5L6.5 15h2.1l.9-2.2zM12 5.98L13.87 11h-3.74L12 5.98z" />
-											</svg>
-										</div>
+									<div className="flex w-full items-center">
+										<div>{ FieldIcon ? <FieldIcon /> : null }</div>
 										<span className=" ml-4 truncate">{ field.label }</span>
 										<div className="flex items-center h-6 px-2 bg-gray-200 rounded-sm ml-auto hover:bg-gray-400">
 											<span className="text-xs font-mono">{ field.name }</span>
 											<ClipboardCopy text={ field.name } />
 										</div>
-										{ shouldDisplayMoveButtons
-											? (
-												<div
+									</div>
+									{ null === parentField && 'repeater' === field.control
+										? (
+											<FieldsGrid
```

Comment:
A repeater also has a grid of fields. So it recursively calls this component:

![fields-giasd](https://user-images.githubusercontent.com/4063887/102319623-cd509800-3f40-11eb-89b5-8963d21e94cd.png)